### PR TITLE
Add unit-tests for vp8 packetization and depacketization

### PR DESCRIPTION
**Pull request by t2-g**
*Description of changes:*
This PR adds unit-tests for vp8 packetization and depacketization. There is also acbug fix in the vp8_depacketizer.c file.

*Test Steps:*

```
git submodule update --init --recursive --checkout test/CMock
cmake -S test/unit-test -B build/ -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Debug -DBUILD_CLONE_SUBMODULES=ON -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -Werror -DNDEBUG -DLIBRARY_LOG_LEVEL=LOG_DEBUG'
make -C build all
cd build
ctest -E system --output-on-failure
make coverage
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/vp8/vp8_utest.c [hunk 1]
```diff
@@ -301,11 +303,168 @@ void test_VP8_Packetizer_TL0PICIDX( void )
...
+    TEST_ASSERT_EQUAL( VP8_RESULT_OK,
+                       result );
+
+    pkt.pPacketData = &( packetBuffer[ 0 ] );
+    pkt.packetDataLength = PACKET_BUFFER_LENGTH;
+
+    result = VP8Packetizer_GetPacket( &( ctx ),
+                                      &( pkt ) );
+
+    while( result != VP8_RESULT_NO_MORE_PACKETS )
```
Comment (ActoryOu) [inline test/unit-test/vp8/vp8_utest.c:370]: Can we calculate the number of packet first? Then use the calculated number to check the expected result. For example, if we expect to have 3 packets from this frame, we can expect to have 3 OK returned and 1 NO_MORE_PACKETS at the end.

File: test/unit-test/vp8/vp8_utest.c [hunk 2]
```diff
@@ -301,11 +303,168 @@ void test_VP8_Packetizer_TL0PICIDX( void )
...
+    TEST_ASSERT_EQUAL( VP8_RESULT_OK,
+                       result );
+
+    pkt.pPacketData = &( packetBuffer[0] );
+    pkt.packetDataLength = PACKET_BUFFER_LENGTH;
+
+    result = VP8Packetizer_GetPacket( &ctx,
+                                      &pkt );
+
+    while( result != VP8_RESULT_NO_MORE_PACKETS )
```
Comment (ActoryOu) [inline test/unit-test/vp8/vp8_utest.c]: same here

File: test/unit-test/vp8/vp8_utest.c [hunk 3]
```diff
@@ -432,15 +592,153 @@ void test_VP8_Packetizer_AllProperties( void )
...
+
+    result = VP8Packetizer_Init( &( ctx ),
+                                 &( frame ) );
+
+    TEST_ASSERT_EQUAL( VP8_RESULT_OK,
+                       result );
+
+    pkt.pPacketData = &( packetBuffer[ 0 ] );
+    pkt.packetDataLength = PACKET_BUFFER_LENGTH;
+    ctx.payloadDescLength = PACKET_BUFFER_LENGTH + 1;
```
Comment (ActoryOu) [inline test/unit-test/vp8/vp8_utest.c]: I think this is setting the needed memory size fo a packet, is it?

**Review by ActoryOu — COMMENTED** [review]
Could you execute the formatting with uncrustify once again?

File: test/unit-test/vp8/vp8_utest.c (hunk 3 above)
Comment (t2-g) [reply test/unit-test/vp8/vp8_utest.c]: Yes, so the actual payload length is 1 more than what can be packetized which gives us "out of memory" output.

**Review by ActoryOu — DISMISSED** [review]

**Review by aggarg — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

**Pull request merged**